LeMaterial (#2523)
huggingface/blog: lematerial.md

**LeMaterial** addresses these challenges by unifying and standardizing data from three major databases—Materials Project, Alexandria, and OQMD—into a high-quality resource with consistent and systematic properties. The elemental composition treemap below highlights the value of this integration, showcasing how we increase the scope of existing datasets, like Materials Project, which are biased toward specific material types, such as battery materials (Li, O, P) or oxides.

![LeMat_Bulk_unique_materials.png](TODO LINK)

*LeMat-BulkUnique treemap*

![mp_treemap.png](TODO LINK)

*MaterialProject treemap*

> Add a quote from Meta commenting on this dataset
> 

## Achieving a clean, unified and standardized dataset

`LeMat-Bulk` is more than a large-scale merged dataset with a permissive license (CC-4-BY). With its 6.7M entries with consistent properties, it represents a foundational step towards creating a curated and standardized open ecosystem for material science, designed to simplify research workflows and improve data quality. Here's a closer view of what is looks like and how it was constructed.

> ![Screenshot 2024-12-09 at 06.45.32.png](TODO LINK)

Reviewer: could we add an iframe instead of a screenshot here?
<iframe
  src=["https://huggingface.co/datasets/LeMaterial/LeMat-Bulk/embed/viewer/compatible_pbe/train"](https://huggingface.co/datasets/LeMaterial/LeMat-Bulk/embed/viewer/compatible_pbe/train)
  frameborder="0"
  width="100%"
  height="560px"
></iframe>

Reviewer 2: @lvwerra could you add this in a next commit please ?

---
title: "LeMaterial: an open source initiative to accelate materials discovery and research" 
thumbnail: /blog/assets/lematerial/thumbnail_lematerial.png
authors:
- user: AlexDuvalinho
  guest: true
- user: lritchie
  guest: true
- user: msiron
  guest: true
- user: inelgnu
  guest: true
- user: etiennedufayet
  guest: true
- user: amandinerossello
  guest: true
- user: Ramlaoui
  guest: true
- user: IAMJB
- user: lvwerra
- user: thomwolf
---

# LeMaterial: an open source initiative to accelerate materials discovery and research

> Today, we are thrilled to announce the launch of **LeMaterial**, an open-source collaborative project led by [*Entalpic*](https://entalpic.ai/) and [*Hugging Face*](https://huggingface.co/). LeMaterial aims to simplify and accelerate materials research, making it easier to train ML models, to identify novel materials and to explore chemical spaces. ⚛️🤗

Reviewer 3: ```suggestion
Today, we are thrilled to announce the launch of **LeMaterial**, an open-source collaborative project led by [*Entalpic*](https://entalpic.ai/) and [*Hugging Face*](https://huggingface.co/). LeMaterial aims to simplify and accelerate materials research, making it easier to train ML models, identify novel materials and explore chemical spaces. ⚛️🤗
```

Reviewer 2: OK

As a first step, we are releasing a dataset called `LeMat-Bulk`, which unifies, cleans and standardizes the most prominent material datasets, including [Materials Project](https://next-gen.materialsproject.org/), [Alexandria](https://alexandria.icams.rub.de/) and [OQMD](https://oqmd.org/) — giving rise to a single harmonized data format with **6.7M entries** and **7 materials properties.**

## Why LeMaterial?

The world of materials science, at the intersection of quantum chemistry and machine learning, is brimming with opportunity — from brighter LEDs, to electro-chemical batteries, more efficient photovoltaic cells and recyclable plastics, the applications are endless. By leveraging machine learning (ML) on large, structured datasets, researchers can perform high-throughput screening and testing of new materials at unprecedented scales, significantly accelerating the discovery cycle of novel compounds with desired properties. In this paradigm, **data becomes the essential fuel powering ML models** that can guide experiments, reduce costs, and unlock breakthroughs faster than ever before.

> However, **this field is hampered by** **fragmented datasets** **that vary in format, parameters, and scope, presenting the following challenges:**

Reviewer 3: Change the bolding here - either do it in one step or select what you highlight

Reviewer 2: Okay, let's replace by 
However, **this field is hampered by fragmented datasets that vary in format, parameters, and scope, presenting the following challenges:**
Member

Reviewer 2: @lvwerra to add in next commit

Author: that's just markdown cosmetic, right? because it will render the same?

**LeMaterial** addresses these challenges by unifying and standardizing data from three major databases—Materials Project, Alexandria, and OQMD—into a high-quality resource with consistent and systematic properties. The elemental composition treemap below highlights the value of this integration, showcasing how we increase the scope of existing datasets, like Materials Project, which are biased toward specific material types, such as battery materials (Li, O, P) or oxides.

<p align="center">
    <img src="https://huggingface.co/datasets/LeMaterial/admin/resolve/main/MP_LeMatBulk_Combined_Treemap.png" alt="drawing" width="1000"/>
</p>

<p align="center">
  <em>Materials Project and LeMat-BulkUnique treemap</em>
</p>

## Achieving a clean, unified and standardized dataset

> `LeMat-Bulk` is more than a large-scale merged dataset with a permissive license (CC-BY-4.0). With its 6.7M entries with consistent properties, it represents a foundational step towards creating a curated and standardized open ecosystem for material science, designed to simplify research workflows and improve data quality. Here's a closer view of what is looks like and how it was constructed.

Reviewer 2: ```suggestion
`LeMat-Bulk` is more than a large-scale merged dataset with a permissive license (CC-BY-4.0). With its 6.7M entries with consistent properties, it represents a foundational step towards creating a curated and standardized open ecosystem for material science, designed to simplify research workflows and improve data quality. Below is a closer view of what is looks like. To interactively browse through our materials, check out the [Materials Explorer space.](https://huggingface.co/spaces/LeMaterial/materials_explorer)
```

> ![image.png](https://huggingface.co/datasets/LeMaterial/admin/resolve/main/Dataset-viewer-screenshot.png)

Reviewer: ```suggestion
<iframe
  src=["https://huggingface.co/datasets/LeMaterial/LeMat-Bulk/embed/viewer/compatible_pbe/train"](https://huggingface.co/datasets/LeMaterial/LeMat-Bulk/embed/viewer/compatible_pbe/train)
  frameborder="0"
  width="100%"
  height="560px"
></iframe>
```

| v.1.1 | <ul><li><strong>New data</strong>: calculate charge data for many materials in Materials Project database (additional 53k data </li>points), band gaps information, unified energy correction scheme<li><strong>Evaluation</strong>: Code hashing function benchmarks</li><li><strong>New functions</strong>: Material similarity metric and retriever tool of similar structures</li><li><strong>New models</strong>: Equiformerv2 and FAENet trained on the data</li><li><strong>Data validation</strong>: check for compatibility of fields, formatting, etc.</li></ul> | Q1 2025 |

> | Future releases | <li><strong>New data</strong>: release of r2SCAN data from Materials Project </li><li><strong>New surface datasets</strong>: OC20 & OC22 datasets and ability to cover other models than bulk (surface and slab+ads and molecule structures)</li><li><strong>Further harmonisation</strong>: include trajectories across MPTrj, OMat24 in similar format, include relational data in our DB </li> | Q2 2025 |

Reviewer 3: Imo this should go at the very end, under a section going something like "But this is only the beginning! Here are the future releases planned, call to contribution"

Reviewer 2: @clefourrier Those are the next steps only for the dataset part, this is why we added it here and not at the very end

> 

Reviewer 2: ```suggestion
**We offer different datasets and subsets,** enabling tailored workflows for researchers depending on their needs (consistent calculations, deduplicating materials, or comprehensive exploration):

- **Compatibility:** these subsets only provides calculations which are compatible to mix. This is available in 3 functionals today (PBE, PBESol and SCAN)
- **Non-compatible:** this subset provides all materials not included in the compatibility subsets.
- [**LeMat-BulkUnique**](https://huggingface.co/datasets/LeMaterial/LeMat-BulkUnique) : this dataset split provides de-duplicated material using our structure fingerprint algorithm. It is available in 3 subsets, for PBE, PBESol, and SCAN functionals. More Details on the dataset can be found on [🤗Hugging Face](https://huggingface.co/datasets/LeMaterial/LeMat-BulkUnique)
```

Reviewer 2: @clefourrier Can you add this part in the last version? Missing from the Notion version

## Integrating a well-benchmarked materials fingerprint

Beside building this standardized dataset, one of the key contribution of LeMaterial is to propose a definition of a material fingerprint through a **hashing function** that assigns a unique identifier to each material.

> Current approaches to identifying a material as novel relative to a database have predominantly relied on similarity metrics, which necessitate a combinatorial effort to screen the existing database for novelty. **To provide for faster detection of novelty in a dataset, Entalpic proposes a hashing method to compute the fingerprint of a material. 

Reviewer 3: ```suggestion
Current approaches to identifying a material as novel relative to a database have predominantly relied on similarity metrics, which necessitate a combinatorial effort to screen the existing database for novelty. To provide faster novelty detection in a dataset, Entalpic introduces a hashing method to compute the fingerprint of a material.
```

Reviewer 2: lgtm

![image.png](https://huggingface.co/datasets/LeMaterial/admin/resolve/main/Hash%20Function.jpeg)
<blockquote>Above is a breakdown of the fingerprinting. We use a bonding algorithm (e.g. EconNN) on the crystal structure to extract a graph, on which we then compute the Weisfeiler-Lehman algorithm to get a hash. This hash is combined with composition and space group information to create the material fingerprint.</blockquote>

**Our fingerprinting approach offers several benefits**: 

> - Quickly identifying whether a material is novel or already cataloged.

Reviewer 3: ```suggestion
- Quickly identifying whether a material is novel or already catalogued.
```

Reviewer 2: we keep cataloged as American English, we spelled everything as American English

- Ensuring the dataset is free from duplicates and inconsistencies.
- Allowing to connect materials between datasets.
- Supporting more efficient calculations for thermodynamic properties, such as energy above the hull.

Below lies a comparison of our hash function with the [StructureMatcher](https://pymatgen.org/pymatgen.analysis.html#pymatgen.analysis.structure_matcher.StructureMatcher:~:text=%3D%20True%2C-,attempt_supercell,-%3A%20bool%20%3D) of Pymatgen, to find all duplicates of a dataset. The experiment was run on two datasets having very different  structures.

> For the hash function task, **almost all of the task time was dedicated to calculating the hashes**. Their comparison time is negligible. For StructureMatcher, the vast majority of the task time was spent **comparing pairs of structures**. Their building time is negligible. 

Reviewer 3: ```suggestion
When using our method, **almost all of the task time was dedicated to calculating material hashes**; the follow-up comparison step is negligible time-wise. When using `StructureMatcher`, the vast majority of the task time was spent **comparing pairs of structures**; building said structures is negligible time-wise.
```

Reviewer 2: lgtm

| **Dataset** | **Number of structures** | Task time for the hash function (parallelized on 12 CPUs) | Task time for StructureMatcher (parallelized on 64 CPUs) |
| --- | --- | --- | --- |
| Carbon-24 | 10,153 | 100 seconds | 17 hours |

> | MPTS-52 | 40,476 | 330 seconds | 4.9 hours |

Reviewer 3: Add a small comment that we're order of magnitudes faster

Additionally, we are planning on releasing **a set of well-curated benchmarks** to evaluate the validity of our hashing function. For instance, we investigated:

- If distinct materials lead to different hashes based on material identification tags across existing databases
- Whether adding small noises or applying symmetry operations to a material leads to the same hash
- If materials sharing the same hash, across or within databases, could indeed be the same material — with manual and DFT checks
- How fast and accurate our hash is compared to Pymatgen's StructureMatcher on existing databases

<blockquote>

>   <strong> 🤗 Call to the community:</strong> the aim is not to position our fingerprint method as the single solution to de-duplicate materials databases and find novel materials, but rather to foster discussion around this question. One current limitation of our hash, is that it does not address disordered structures. Hence, we would like to push the community towards achieving a consensus, while proposing a relatively simple and efficient fingerprint method in the meantime.</blockquote>

Reviewer 3: ```suggestion
<strong> 🤗 Call to the community:</strong> our aim is not to position this fingerprint method as the single solution to de-duplicate materials databases and find novel materials, but rather to foster discussion around this question. One current limitation of this hashing technique is that it does not cover disordered structures; we would like to push the community towards finding a consensus, while proposing a relatively simple and efficient fingerprint method in the meantime.</blockquote>

Reviewer 2: changes lgtm

## LeMaterial in Action: applications and impact

LeMaterial aims to be a community-driven initiative that gathers machine learning models, visualization tools, large & curated datasets, handy toolkits, etc. It is designed to be practical and flexible, enabling a wide range of applications, such as:

> - **Exploring extended phase diagrams** ([link to Hugging Face](https://huggingface.co/spaces/LeMaterial/phase_diagram)), constructed with a broader dataset, to analyze chemical spaces in greater detail. Combining larger datasets means that we can provide a finer resolution of material stability in a given compositional space:

Reviewer 3: ```suggestion
- **Exploring extended phase diagrams** ([link to our explorer](https://huggingface.co/spaces/LeMaterial/phase_diagram)), constructed with a broader dataset, to analyze chemical spaces in greater detail. Combining larger datasets means that we can provide a finer resolution of material stability in a given compositional space:

---
title: "LeMaterial: an open source initiative to accelate materials discovery and research" 
thumbnail: /blog/assets/lematerial/thumbnail_lematerial.png
authors:
- user: AlexDuvalinho
  guest: true
- user: lritchie
  guest: true
- user: msiron
  guest: true
- user: inelgnu
  guest: true
- user: etiennedufayet
  guest: true
- user: amandinerossello
  guest: true
- user: Ramlaoui
  guest: true
- user: IAMJB
- user: lvwerra
- user: thomwolf
---

> 

Author: ```suggestion

![logos](/blog/assets/lematerial/thumbnail_lematerial.png)

```

# LeMaterial: an open source initiative to accelerate materials discovery and research

> 

Author: @lritchie-entalpic do we want to add the banner back at the start of blog? i think the export removed it.

Reviewer 2: Yes please would be great to add the banner
In the Notion version, was just above the "Why LeMaterial?" Title and not below the initial title. Great if we can add it here

As a first step, we are releasing a dataset called `LeMat-Bulk`, which unifies, cleans and standardizes the most prominent material datasets, including [Materials Project](https://next-gen.materialsproject.org/), [Alexandria](https://alexandria.icams.rub.de/) and [OQMD](https://oqmd.org/) — giving rise to a single harmonized data format with **6.7M entries** and **7 materials properties.**

## Why LeMaterial?

The world of materials science, at the intersection of quantum chemistry and machine learning, is brimming with opportunity — from brighter LEDs, to electro-chemical batteries, more efficient photovoltaic cells and recyclable plastics, the applications are endless. By leveraging machine learning (ML) on large, structured datasets, researchers can perform high-throughput screening and testing of new materials at unprecedented scales, significantly accelerating the discovery cycle of novel compounds with desired properties. In this paradigm, **data becomes the essential fuel powering ML models** that can guide experiments, reduce costs, and unlock breakthroughs faster than ever before.

> However, **this field is hampered by** **fragmented datasets** **that vary in format, parameters, and scope, presenting the following challenges:**

Author: ```suggestion
However, **this field is hampered by fragmented datasets that vary in format, parameters, and scope, presenting the following challenges:**
```

does that look good @lritchie-entalpic ?

Reviewer 2: yes perfect thanks @lvwerra

| v.1.1 | <ul><li><strong>New data</strong>: calculate charge data for many materials in Materials Project database (additional 53k data </li>points), band gaps information, unified energy correction scheme<li><strong>Evaluation</strong>: Code hashing function benchmarks</li><li><strong>New functions</strong>: Material similarity metric and retriever tool of similar structures</li><li><strong>New models</strong>: Equiformerv2 and FAENet trained on the data</li><li><strong>Data validation</strong>: check for compatibility of fields, formatting, etc.</li></ul> | Q1 2025 |
| Future releases | <li><strong>New data</strong>: release of r2SCAN data from Materials Project </li><li><strong>New surface datasets</strong>: OC20 & OC22 datasets and ability to cover other models than bulk (surface and slab+ads and molecule structures)</li><li><strong>Further harmonisation</strong>: include trajectories across MPTrj, OMat24 in similar format, include relational data in our DB </li> | Q2 2025 |

> **We offer different datasets and subsets,** enabling tailored workflows for researchers depending on their needs (consistent calculations, deduplicating materials, or comprehensive exploration):

Author: ```suggestion

**We offer different datasets and subsets,** enabling tailored workflows for researchers depending on their needs (consistent calculations, deduplicating materials, or comprehensive exploration):
```
